pkg/sql/schemachanger/scpb/elements.proto

```diff
@@ -279,6 +279,7 @@ message SecondaryIndex {
 message TemporaryIndex {
   Index embedded_index = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
   bool is_using_secondary_encoding = 2;
+  Expression embedded_expr = 3 [(gogoproto.nullable) = true];
```

Comment:
nit: add some comments around this new field
nit: if you really intend this to be embedded (via `(gogoproto.embed) = true`), then consider change the name?